### DISCUSSION TITLE
ENOSPC: no space left on device, write

### DISCUSSION THREAD

**Discussion by masapasa** [vercel/community#2795] [Help]
### Summary

Fastapi with langchain on the backend and nextjs on the front end. i am trying to deploy to deploy this repo https://github.com/masapasa/chatdoc

### Example

https://github.com/masapasa/chatdoc

### Steps to Reproduce

51:6  Warning: React Hook useEffect has a missing dependency: 'setQuestion'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
--
18:46:31.388 |  
18:46:31.388 | ./app/components/Footer/index.tsx
18:46:31.389 | 11:11  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
18:46:31.390 | 19:11  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
18:46:31.390 |  
18:46:31.390 | ./app/explore/page.tsx
18:46:31.390 | 41:6  Warning: React Hook useEffect has a missing dependency: 'axiosInstance'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
18:46:31.390 |  
18:46:31.391 | ./app/upload/components/Crawler/hooks/useCrawler.ts
18:46:31.391 | 58:6  Warning: React Hook useCallback has a missing dependency: 'axiosInstance'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
18:46:31.391 |  
18:46:31.391 | ./app/upload/components/FileUploader/hooks/useFileUploader.ts
18:46:31.391 | 41:5  Warning: React Hook useCallback has a missing dependency: 'axiosInstance'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps
18:46:31.391 |  
18:46:31.391 | info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
18:46:33.933 | - info Collecting page data...
18:46:44.042 | - info Generating static pages (0/13)
18:46:44.149 | - info Generating static pages (3/13)
18:46:44.182 | - warn Entire page /login deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /login
18:46:44.187 | - warn Entire page /logout deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /logout
18:46:44.226 | - warn Entire page /signup deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /signup
18:46:44.227 | - info Generating static pages (6/13)
18:46:44.234 | - warn Entire page /chat deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /chat
18:46:44.247 | - warn Entire page / deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /
18:46:44.258 | - warn Entire page /config deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /config
18:46:44.272 | - info Generating static pages (9/13)
18:46:44.282 | - warn Entire page /explore deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /explore
18:46:44.303 | - warn Entire page /upload deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /upload
18:46:44.325 | - warn Entire page /user deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /user
18:46:44.337 | - info Generating static pages (13/13)
18:46:44.350 | - info Finalizing page optimization...
18:46:44.353 |  
18:46:44.406 | Route (app)                                Size     First Load JS
18:46:44.406 | ┌ λ /                                      0 B                0 B
18:46:44.406 | ├ λ /chat                                  37.9 kB         209 kB
18:46:44.407 | ├ λ /config                                10.3 kB         129 kB
18:46:44.407 | ├ λ /explore                               15.2 kB         199 kB
18:46:44.407 | ├ ○ /favicon.ico                           0 B                0 B
18:46:44.407 | ├ λ /login                                 0 B                0 B
18:46:44.407 | ├ λ /logout                                0 B                0 B
18:46:44.407 | ├ λ /signup                                0 B                0 B
18:46:44.407 | ├ λ /upload                                10.8 kB         184 kB
18:46:44.407 | └ λ /user                                  97.7 kB         234 kB
18:46:44.407 | + First Load JS shared by all              90.7 kB
18:46:44.407 | ├ chunks/139-3fa1ce7dfff2c17b.js         24.7 kB
18:46:44.407 | ├ chunks/2443530c-da70b571b1f7722c.js    50.5 kB
18:46:44.407 | ├ chunks/434-a0bb612bd29cffc2.js         13.2 kB
18:46:44.407 | ├ chunks/b9eae562-5b831bcb1a0748a1.js    348 B
18:46:44.407 | ├ chunks/main-app-050734bcccde684a.js    221 B
18:46:44.407 | └ chunks/webpack-ea58be35696dac38.js     1.79 kB
18:46:44.408 |  
18:46:44.408 | Route (pages)                              Size     First Load JS
18:46:44.408 | ┌ ○ /404                                   180 B          90.5 kB
18:46:44.408 | └ ○ /about                                 733 B            91 kB
18:46:44.408 | + First Load JS shared by all              90.3 kB
18:46:44.408 | ├ chunks/main-e4d079f5808e32a4.js        88.3 kB
18:46:44.408 | ├ chunks/pages/_app-dec1a83570915fb0.js  195 B
18:46:44.408 | └ chunks/webpack-ea58be35696dac38.js     1.79 kB
18:46:44.408 |  
18:46:44.408 | ƒ Middleware                               98.7 kB
18:46:44.408 |  
18:46:44.408 | λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
18:46:44.408 | ○  (Static)  automatically rendered as static HTML (uses no initial props)
18:46:44.408 |  
18:46:47.789 | Traced Next.js server files in: 2.688s
18:46:54.737 | Created all serverless functions in: 6.947s
18:46:54.804 | Collected static files (public/, static/, .next/static): 16.981ms
18:46:54.816 | Installing required dependencies...
18:47:26.285 | Installing required dependencies...
18:47:57.646 | Installing required dependencies...
18:48:28.711 | Installing required dependencies...
18:49:00.979 | Installing required dependencies...
18:49:32.299 | Installing required dependencies...
18:50:03.670 | Installing required dependencies...
18:50:34.890 | Installing required dependencies...
18:51:05.886 | Installing required dependencies...
18:51:37.202 | Installing required dependencies...
18:52:08.447 | Installing required dependencies...
18:52:40.748 | Installing required dependencies...
18:53:12.024 | Installing required dependencies...
18:53:43.610 | Installing required dependencies...
18:54:14.878 | Installing required dependencies...
18:54:46.026 | Installing required dependencies...
18:55:18.627 | Installing required dependencies...
18:55:49.753 | Installing required dependencies...
18:56:21.425 | Installing required dependencies...
18:56:52.665 | Installing required dependencies...
18:57:23.729 | Installing required dependencies...
18:57:55.103 | Installing required dependencies...
18:58:26.281 | Installing required dependencies...
18:58:58.695 | Installing required dependencies...
18:59:30.157 | Installing required dependencies...
19:00:01.391 | Installing required dependencies...
19:00:32.825 | Installing required dependencies...
19:01:04.648 | Error: ENOSPC: no space left on device, write
19:01:04.665 | Error: ENOSPC: no space left on device, write
19:01:06.540 | ENOSPC: ENOSPC: no space left on device, write

**Comment by masapasa** [comment]
Error: Command failed: pip3.9 install --disable-pip-version-check --target . werkzeug==1.0.1
--
19:55:36.272 | ERROR: Could not install packages due to an OSError: [Errno 28] No space left on device
19:55:36.272 |  
19:55:36.272 |  
19:55:36.272 | Collecting werkzeug==1.0.1
19:55:36.272 | Using cached Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)
19:55:36.272 |  
19:55:37.979 | 1: Command failed: pip3.9 install --disable-pip-version-check --target . werkzeug==1.0.1ERROR: Could not install packages due to an OSError: [Errno 28] No space left on deviceCollecting werkzeug==1.0.1  Using cached Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)

**Comment by GustavoContreiras** [comment]
I'm having the same ENOSPC issue with Python.
vercel.json:
```
{
    "builds": [
        {
            "src": "main.py",
            "use": "@vercel/python",
            "config": { "newPipeline": true }
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "main.py"
        }
    ]
}
```

requirements.txt:
```
openai
faiss-cpu
python-dotenv
fastapi
langchain[llms]
bs4
sentence_transformers
transformers[torch]
torch
huggingface_hub
xformers
tiktoken
gpt4all
wheel
pypdf
boto3
pgvector
psycopg2-binary
```
